<a href="https://cognitiveclass.ai/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDL0101ENSkillsNetwork20718188-2022-01-01"><img src = "https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/Logos/organization_logo/organization_logo.png" width = 400> </a>

<h1 align=center><font size = 5>Regression Models with Keras</font></h1>


<h2>Regression Models with Keras</h2>
  


## Table of Contents

<div class="alert alert-block alert-info" style="margin-top: 20px">

<font size = 3>

1.  <a href="https://#item31">Download and Clean Dataset</a>
2.  <a href="https://#item32">Import Keras</a>
3.  <a href="https://#item33">Build a Neural Network</a>
4.  <a href="https://#item34">Train and Test the Network</a>

</font>
</div>


<a id="item31"></a>


## Download and Clean Dataset


Let's start by importing the <em>pandas</em> and the Numpy libraries.


In [ ]:
# All Libraries required for this lab are listed below. The libraries pre-installed on Skills Network Labs are commented. 
# If you run this notebook on a different environment, e.g. your desktop, you may need to uncomment and install certain libraries.

#!pip install numpy==1.21.4
#!pip install pandas==1.3.4
#!pip install keras==2.1.6

In [1]:
import pandas as pd
import numpy as np

We will be playing around with the same dataset that we used in the videos.

<strong>The dataset is about the compressive strength of different samples of concrete based on the volumes of the different ingredients that were used to make them. Ingredients include:</strong>

<strong>1. Cement</strong>

<strong>2. Blast Furnace Slag</strong>

<strong>3. Fly Ash</strong>

<strong>4. Water</strong>

<strong>5. Superplasticizer</strong>

<strong>6. Coarse Aggregate</strong>

<strong>7. Fine Aggregate</strong>


Let's download the data and read it into a <em>pandas</em> dataframe.


In [2]:
concrete_data = pd.read_csv('https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DL0101EN/labs/data/concrete_data.csv')
concrete_data.head()

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  Strength  
0            1040.0           676.0   28     79.99  
1            1055.0           676.0   28     61.89  
2             932.0           594.0  270     40.27  
3             932.0           594.0  365     41.05  
4             978.4           825.5  360     44.30

So the first concrete sample has 540 cubic meter of cement, 0 cubic meter of blast furnace slag, 0 cubic meter of fly ash, 162 cubic meter of water, 2.5 cubic meter of superplaticizer, 1040 cubic meter of coarse aggregate, 676 cubic meter of fine aggregate. Such a concrete mix which is 28 days old, has a compressive strength of 79.99 MPa.


#### Let's check how many data points we have.


In [3]:
concrete_data.shape

(1030, 9)

So, there are approximately 1000 samples to train our model on. Because of the few samples, we have to be careful not to overfit the training data.


Let's check the dataset for any missing values.


In [4]:
concrete_data.describe()

Cement  Blast Furnace Slag      Fly Ash        Water  \
count  1030.000000         1030.000000  1030.000000  1030.000000   
mean    281.167864           73.895825    54.188350   181.567282   
std     104.506364           86.279342    63.997004    21.354219   
min     102.000000            0.000000     0.000000   121.800000   
25%     192.375000            0.000000     0.000000   164.900000   
50%     272.900000           22.000000     0.000000   185.000000   
75%     350.000000          142.950000   118.300000   192.000000   
max     540.000000          359.400000   200.100000   247.000000   

       Superplasticizer  Coarse Aggregate  Fine Aggregate          Age  \
count       1030.000000       1030.000000     1030.000000  1030.000000   
mean           6.204660        972.918932      773.580485    45.662136   
std            5.973841         77.753954       80.175980    63.169912   
min            0.000000        801.000000      594.000000     1.000000   
25%            0.000000        932.000000      730.950000     7.000000   
50%            6.400000        968.000000      779.500000    28.000000   
75%           10.200000       1029.400000      824.000000    56.000000   
max           32.200000       1145.000000      992.600000   365.000000   

          Strength  
count  1030.000000  
mean     35.817961  
std      16.705742  
min       2.330000  
25%      23.710000  
50%      34.445000  
75%      46.135000  
max      82.600000

In [5]:
concrete_data.isnull().sum()

Cement                0
Blast Furnace Slag    0
Fly Ash               0
Water                 0
Superplasticizer      0
Coarse Aggregate      0
Fine Aggregate        0
Age                   0
Strength              0
dtype: int64

The data looks very clean and is ready to be used to build our model.


#### Split data into predictors and target


The target variable in this problem is the concrete sample strength. Therefore, our predictors will be all the other columns.


In [6]:
concrete_data_columns = concrete_data.columns

predictors = concrete_data[concrete_data_columns[concrete_data_columns != 'Strength']] # all columns except Strength
target = concrete_data['Strength'] # Strength column

<a id="item2"></a>


Let's do a quick sanity check of the predictors and the target dataframes.


In [7]:
predictors.head()

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  
0            1040.0           676.0   28  
1            1055.0           676.0   28  
2             932.0           594.0  270  
3             932.0           594.0  365  
4             978.4           825.5  360

In [50]:
target.head()

0    79.99
1    61.89
2    40.27
3    41.05
4    44.30
Name: Strength, dtype: float64

<a id="item1"></a>


<a id='item32'></a>


## Regression with Keras


In [51]:
import keras

In [52]:
from keras.models import Sequential
from keras.layers import Dense

<a id='item33'></a>


## Build a Neural Network


Let's define a function that defines our regression model for us so that we can conveniently call it to create our model.


In [53]:
# define regression model, generic function
def regression_model(num_hidden_layer,num_node,n_cols):
    # create model
    model = Sequential()
    
    # create hidden layers
    for index in range(num_hidden_layer):
        if index==0:
            model.add(Dense(num_node[index], activation='relu', input_shape=(n_cols,)))
        else:
            model.add(Dense(num_node[index], activation='relu'))
            
    model.add(Dense(1))
    
    # compile model
    model.compile(optimizer='adam', loss='mean_squared_error')
    return model

The above function is a generic function to create a model that has N hidden layers, each of N hidden units.

inputs : num_hidden_layer,num_node,n_cols

## A. Build a baseline model (5 marks) 
Use the Keras library to build a neural network with the following:

- One hidden layer of 10 nodes, and a ReLU activation function

- Use the adam optimizer and the mean squared error  as the loss function.

1. Randomly split the data into a training and test sets by holding 30% of the data for testing. You can use the train_test_splithelper function from Scikit-learn.

2. Train the model on the training data using 50 epochs.

3. Evaluate the model on the test data and compute the mean squared error between the predicted concrete strength and the actual concrete strength. You can use the mean_squared_error function from Scikit-learn.

4. Repeat steps 1 - 3, 50 times, i.e., create a list of 50 mean squared errors.

5. Report the mean and the standard deviation of the mean squared errors.


In [54]:
#import train_test_split
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
import numpy as np

print("predictors ", predictors.shape)
print("target ", target.shape)

# splitting predictors,target  // 30% of the data for testing
predictors_train,predictors_test,target_train,target_test = train_test_split(predictors,target,test_size = 0.3,random_state=4)

print("predictors_train ", predictors_train.shape)
print("predictors_test ", predictors_test.shape)

# number of predictors_train_cols
predictors_train_cols = predictors_train.shape[1]

# number of predictors_test_cols
predictors_test_cols = predictors_test.shape[1]


predictors  (1030, 8)
target  (1030,)
predictors_train  (721, 8)
predictors_test  (309, 8)


In [55]:
# list of 50 mean squared errors. repeat 1-3
N=50
MS_Error  = np.zeros((N))
for ind in range(N):
    predictors_train,predictors_test,target_train,target_test = train_test_split(predictors,target,test_size = 0.3,random_state=4)
    # One hidden layer of 10 nodes, and a ReLU activation function
    # build the model
    model = regression_model(1,[10],predictors_train_cols)
    # Train the model on the training data using 50 epochs.
    model.fit(predictors_train, target_train, validation_split=0.3, epochs=50, verbose=2)
    # Evaluate the model on the test data 
    predicted_target_values = model.predict(predictors_test)
    # compute the mean squared error between the predicted concrete 
    # strength and the actual concrete strength
    # predicted concrete strength  : predicted_target_values
    # actual concrete strength  :  target_test
    MS_Error[ind] = mean_squared_error(target_test, predicted_target_values)


Epoch 1/50
16/16 - 1s - loss: 80674.3984 - val_loss: 58138.6445 - 537ms/epoch - 34ms/step
Epoch 2/50
16/16 - 0s - loss: 45768.4375 - val_loss: 32401.6523 - 34ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 25396.6230 - val_loss: 17931.2930 - 30ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 14080.5801 - val_loss: 9706.9629 - 29ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 7622.1929 - val_loss: 5163.9624 - 30ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 4089.7002 - val_loss: 2720.0164 - 31ms/epoch - 2ms/step
Epoch 7/50
16/16 - 0s - loss: 2207.0771 - val_loss: 1485.0912 - 29ms/epoch - 2ms/step
Epoch 8/50
16/16 - 0s - loss: 1294.8185 - val_loss: 907.6058 - 31ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 875.5433 - val_loss: 673.0352 - 30ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 710.6160 - val_loss: 581.6758 - 30ms/epoch - 2ms/step
Epoch 11/50
16/16 - 0s - loss: 644.7010 - val_loss: 545.7256 - 30ms/epoch - 2ms/step
Epoch 12/50
16/16 - 0s - loss: 613.3860 - val_loss

Epoch 46/50
16/16 - 0s - loss: 550.4352 - val_loss: 691.6880 - 29ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 532.6846 - val_loss: 664.8997 - 28ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 516.1108 - val_loss: 646.0095 - 29ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 499.4700 - val_loss: 627.1815 - 27ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 484.3912 - val_loss: 606.3673 - 26ms/epoch - 2ms/step
10/10 [==============================] - 0s 325us/step
Epoch 1/50
16/16 - 0s - loss: 43122.6289 - val_loss: 25799.2051 - 275ms/epoch - 17ms/step
Epoch 2/50
16/16 - 0s - loss: 16335.4805 - val_loss: 9168.6299 - 27ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 6353.8945 - val_loss: 5404.2095 - 27ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 4631.3760 - val_loss: 4824.8755 - 27ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 4134.6421 - val_loss: 4215.3994 - 27ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 3622.9917 - val_loss: 3715.3914 - 27ms/epoch - 2ms/step

16/16 - 0s - loss: 356.7019 - val_loss: 381.1511 - 28ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 345.4379 - val_loss: 363.1305 - 28ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 325.4433 - val_loss: 349.9271 - 29ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 314.2859 - val_loss: 332.1641 - 30ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 301.7942 - val_loss: 319.4955 - 42ms/epoch - 3ms/step
Epoch 45/50
16/16 - 0s - loss: 291.5002 - val_loss: 309.2651 - 24ms/epoch - 1ms/step
Epoch 46/50
16/16 - 0s - loss: 283.0367 - val_loss: 299.9198 - 33ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 275.1288 - val_loss: 291.3149 - 20ms/epoch - 1ms/step
Epoch 48/50
16/16 - 0s - loss: 267.6928 - val_loss: 282.3473 - 30ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 260.1715 - val_loss: 276.0691 - 34ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 255.2069 - val_loss: 268.1484 - 23ms/epoch - 1ms/step
10/10 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - 

Epoch 35/50
16/16 - 0s - loss: 297.9291 - val_loss: 302.5244 - 16ms/epoch - 977us/step
Epoch 36/50
16/16 - 0s - loss: 279.1694 - val_loss: 285.1970 - 30ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 261.8658 - val_loss: 276.3797 - 34ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 246.2900 - val_loss: 256.5476 - 31ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 233.8069 - val_loss: 248.8286 - 21ms/epoch - 1ms/step
Epoch 40/50
16/16 - 0s - loss: 223.4777 - val_loss: 235.8115 - 28ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 213.8317 - val_loss: 229.4700 - 29ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 203.6187 - val_loss: 219.6597 - 28ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 195.9352 - val_loss: 212.7293 - 28ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 189.0213 - val_loss: 207.4360 - 31ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 181.6889 - val_loss: 199.4217 - 31ms/epoch - 2ms/step
Epoch 46/50
16/16 - 0s - loss: 176.3306 - val_loss: 196.6705 - 

Epoch 30/50
16/16 - 0s - loss: 672.9584 - val_loss: 673.1013 - 25ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 655.1674 - val_loss: 659.0252 - 30ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 640.1478 - val_loss: 642.1338 - 30ms/epoch - 2ms/step
Epoch 33/50
16/16 - 0s - loss: 628.7197 - val_loss: 641.7661 - 20ms/epoch - 1ms/step
Epoch 34/50
16/16 - 0s - loss: 616.9436 - val_loss: 614.8782 - 30ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 595.0392 - val_loss: 596.9792 - 30ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 580.1971 - val_loss: 583.1260 - 30ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 568.5063 - val_loss: 575.4034 - 30ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 553.2820 - val_loss: 556.1205 - 29ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 546.6754 - val_loss: 558.9731 - 21ms/epoch - 1ms/step
Epoch 40/50
16/16 - 0s - loss: 531.8820 - val_loss: 531.1249 - 20ms/epoch - 1ms/step
Epoch 41/50
16/16 - 0s - loss: 517.8243 - val_loss: 518.5361 - 30

Epoch 24/50
16/16 - 0s - loss: 624.2508 - val_loss: 648.5486 - 21ms/epoch - 1ms/step
Epoch 25/50
16/16 - 0s - loss: 598.4958 - val_loss: 626.9884 - 28ms/epoch - 2ms/step
Epoch 26/50
16/16 - 0s - loss: 576.5381 - val_loss: 608.1246 - 45ms/epoch - 3ms/step
Epoch 27/50
16/16 - 0s - loss: 554.5623 - val_loss: 586.5620 - 27ms/epoch - 2ms/step
Epoch 28/50
16/16 - 0s - loss: 533.1663 - val_loss: 569.9565 - 26ms/epoch - 2ms/step
Epoch 29/50
16/16 - 0s - loss: 513.6413 - val_loss: 551.2900 - 44ms/epoch - 3ms/step
Epoch 30/50
16/16 - 0s - loss: 494.2965 - val_loss: 535.8506 - 43ms/epoch - 3ms/step
Epoch 31/50
16/16 - 0s - loss: 477.0442 - val_loss: 518.8224 - 39ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 460.1949 - val_loss: 504.6353 - 22ms/epoch - 1ms/step
Epoch 33/50
16/16 - 0s - loss: 444.2669 - val_loss: 487.8518 - 29ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 431.0305 - val_loss: 476.4908 - 28ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 414.9262 - val_loss: 460.1186 - 28

Epoch 20/50
16/16 - 0s - loss: 676.5985 - val_loss: 770.7792 - 28ms/epoch - 2ms/step
Epoch 21/50
16/16 - 0s - loss: 619.8163 - val_loss: 717.1346 - 28ms/epoch - 2ms/step
Epoch 22/50
16/16 - 0s - loss: 572.5913 - val_loss: 654.0335 - 28ms/epoch - 2ms/step
Epoch 23/50
16/16 - 0s - loss: 538.4398 - val_loss: 620.8265 - 28ms/epoch - 2ms/step
Epoch 24/50
16/16 - 0s - loss: 500.4653 - val_loss: 563.5483 - 29ms/epoch - 2ms/step
Epoch 25/50
16/16 - 0s - loss: 475.0021 - val_loss: 533.8843 - 35ms/epoch - 2ms/step
Epoch 26/50
16/16 - 0s - loss: 454.7587 - val_loss: 505.4467 - 38ms/epoch - 2ms/step
Epoch 27/50
16/16 - 0s - loss: 435.7217 - val_loss: 476.9799 - 29ms/epoch - 2ms/step
Epoch 28/50
16/16 - 0s - loss: 419.5164 - val_loss: 455.2687 - 29ms/epoch - 2ms/step
Epoch 29/50
16/16 - 0s - loss: 404.8716 - val_loss: 439.7751 - 23ms/epoch - 1ms/step
Epoch 30/50
16/16 - 0s - loss: 393.5234 - val_loss: 420.8004 - 28ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 387.3285 - val_loss: 419.0602 - 18

Epoch 14/50
16/16 - 0s - loss: 1461.7927 - val_loss: 1634.9752 - 19ms/epoch - 1ms/step
Epoch 15/50
16/16 - 0s - loss: 1381.9417 - val_loss: 1519.4829 - 30ms/epoch - 2ms/step
Epoch 16/50
16/16 - 0s - loss: 1279.9373 - val_loss: 1422.2942 - 30ms/epoch - 2ms/step
Epoch 17/50
16/16 - 0s - loss: 1201.2080 - val_loss: 1326.5314 - 38ms/epoch - 2ms/step
Epoch 18/50
16/16 - 0s - loss: 1120.3549 - val_loss: 1233.9602 - 30ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 1050.6924 - val_loss: 1158.4978 - 20ms/epoch - 1ms/step
Epoch 20/50
16/16 - 0s - loss: 993.8612 - val_loss: 1099.0051 - 38ms/epoch - 2ms/step
Epoch 21/50
16/16 - 0s - loss: 925.1420 - val_loss: 1033.2650 - 29ms/epoch - 2ms/step
Epoch 22/50
16/16 - 0s - loss: 876.2117 - val_loss: 964.5148 - 33ms/epoch - 2ms/step
Epoch 23/50
16/16 - 0s - loss: 831.8188 - val_loss: 909.2028 - 36ms/epoch - 2ms/step
Epoch 24/50
16/16 - 0s - loss: 788.3913 - val_loss: 860.5730 - 31ms/epoch - 2ms/step
Epoch 25/50
16/16 - 0s - loss: 749.9236 - val_loss:

Epoch 8/50
16/16 - 0s - loss: 5480.9858 - val_loss: 5599.2593 - 32ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 5152.4932 - val_loss: 5282.5107 - 28ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 4869.0508 - val_loss: 4991.1689 - 44ms/epoch - 3ms/step
Epoch 11/50
16/16 - 0s - loss: 4614.0205 - val_loss: 4753.7490 - 20ms/epoch - 1ms/step
Epoch 12/50
16/16 - 0s - loss: 4398.2051 - val_loss: 4510.4399 - 34ms/epoch - 2ms/step
Epoch 13/50
16/16 - 0s - loss: 4193.4224 - val_loss: 4303.1450 - 26ms/epoch - 2ms/step
Epoch 14/50
16/16 - 0s - loss: 4015.6841 - val_loss: 4124.8374 - 28ms/epoch - 2ms/step
Epoch 15/50
16/16 - 0s - loss: 3846.4817 - val_loss: 3947.9966 - 28ms/epoch - 2ms/step
Epoch 16/50
16/16 - 0s - loss: 3693.0396 - val_loss: 3789.9121 - 32ms/epoch - 2ms/step
Epoch 17/50
16/16 - 0s - loss: 3561.1807 - val_loss: 3650.5066 - 23ms/epoch - 1ms/step
Epoch 18/50
16/16 - 0s - loss: 3420.2456 - val_loss: 3513.1792 - 27ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 3296.9092 - va

10/10 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 518432.5000 - val_loss: 433809.9062 - 270ms/epoch - 17ms/step
Epoch 2/50
16/16 - 0s - loss: 386516.7812 - val_loss: 317438.8438 - 18ms/epoch - 1ms/step
Epoch 3/50
16/16 - 0s - loss: 277708.1562 - val_loss: 223502.5000 - 28ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 191653.9375 - val_loss: 151288.8438 - 27ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 127009.9297 - val_loss: 97666.5000 - 26ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 80890.3828 - val_loss: 60832.6133 - 33ms/epoch - 2ms/step
Epoch 7/50
16/16 - 0s - loss: 50084.0977 - val_loss: 36804.9844 - 23ms/epoch - 1ms/step
Epoch 8/50
16/16 - 0s - loss: 30238.4531 - val_loss: 21826.0410 - 28ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 18089.6504 - val_loss: 12893.4307 - 28ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 11018.4971 - val_loss: 7871.8379 - 33ms/epoch - 2ms/step
Epoch 11/50
16/16 - 0s - loss: 7059.8335 - val_loss: 526

Epoch 44/50
16/16 - 0s - loss: 995.7668 - val_loss: 1059.7720 - 28ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 954.6708 - val_loss: 1014.8531 - 32ms/epoch - 2ms/step
Epoch 46/50
16/16 - 0s - loss: 917.9592 - val_loss: 978.3735 - 29ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 883.2357 - val_loss: 943.1486 - 31ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 854.2894 - val_loss: 911.1729 - 31ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 815.1735 - val_loss: 873.9250 - 33ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 787.6771 - val_loss: 843.3043 - 29ms/epoch - 2ms/step
10/10 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 154340.6406 - val_loss: 100993.7266 - 282ms/epoch - 18ms/step
Epoch 2/50
16/16 - 0s - loss: 71014.6016 - val_loss: 39515.9648 - 31ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 23744.7305 - val_loss: 9627.3213 - 33ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 6215.3408 - val_loss: 3910.9548 - 27ms/epoch - 2ms/st

Epoch 39/50
16/16 - 0s - loss: 235.0402 - val_loss: 220.1942 - 28ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 228.9915 - val_loss: 214.2775 - 31ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 223.9801 - val_loss: 209.5183 - 26ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 218.8375 - val_loss: 204.9439 - 30ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 213.8813 - val_loss: 200.5694 - 28ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 209.5587 - val_loss: 195.7330 - 34ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 205.3465 - val_loss: 191.7837 - 25ms/epoch - 2ms/step
Epoch 46/50
16/16 - 0s - loss: 201.0330 - val_loss: 188.4609 - 29ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 197.2444 - val_loss: 185.3135 - 30ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 193.4000 - val_loss: 181.1065 - 32ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 190.6993 - val_loss: 178.6249 - 32ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 187.3420 - val_loss: 174.0487 - 29

Epoch 34/50
16/16 - 0s - loss: 248.9985 - val_loss: 297.7025 - 31ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 240.6095 - val_loss: 288.4338 - 32ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 232.5779 - val_loss: 277.7846 - 28ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 227.7289 - val_loss: 269.0172 - 28ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 219.8032 - val_loss: 260.4928 - 30ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 214.0592 - val_loss: 256.9748 - 29ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 211.7094 - val_loss: 245.7231 - 32ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 203.4000 - val_loss: 240.5027 - 32ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 198.5598 - val_loss: 234.2401 - 30ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 196.5482 - val_loss: 231.8570 - 30ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 191.1211 - val_loss: 222.6322 - 28ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 190.2572 - val_loss: 223.5504 - 47

Epoch 29/50
16/16 - 0s - loss: 131.6150 - val_loss: 132.2076 - 28ms/epoch - 2ms/step
Epoch 30/50
16/16 - 0s - loss: 129.4067 - val_loss: 126.8859 - 29ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 125.4798 - val_loss: 126.2673 - 31ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 123.5781 - val_loss: 123.7266 - 32ms/epoch - 2ms/step
Epoch 33/50
16/16 - 0s - loss: 122.2754 - val_loss: 119.7989 - 29ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 121.0796 - val_loss: 118.9369 - 23ms/epoch - 1ms/step
Epoch 35/50
16/16 - 0s - loss: 119.1233 - val_loss: 120.5710 - 19ms/epoch - 1ms/step
Epoch 36/50
16/16 - 0s - loss: 117.9911 - val_loss: 118.8920 - 34ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 117.3561 - val_loss: 120.3588 - 31ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 118.6351 - val_loss: 114.4396 - 30ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 117.1462 - val_loss: 113.2280 - 34ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 115.4142 - val_loss: 111.9203 - 29

Epoch 23/50
16/16 - 0s - loss: 4274.0425 - val_loss: 4350.0898 - 40ms/epoch - 3ms/step
Epoch 24/50
16/16 - 0s - loss: 4219.5723 - val_loss: 4288.6621 - 34ms/epoch - 2ms/step
Epoch 25/50
16/16 - 0s - loss: 4161.9292 - val_loss: 4235.1187 - 16ms/epoch - 987us/step
Epoch 26/50
16/16 - 0s - loss: 4105.6792 - val_loss: 4184.9268 - 19ms/epoch - 1ms/step
Epoch 27/50
16/16 - 0s - loss: 4048.9099 - val_loss: 4127.1460 - 30ms/epoch - 2ms/step
Epoch 28/50
16/16 - 0s - loss: 3994.2688 - val_loss: 4073.2253 - 30ms/epoch - 2ms/step
Epoch 29/50
16/16 - 0s - loss: 3937.7749 - val_loss: 4021.2739 - 29ms/epoch - 2ms/step
Epoch 30/50
16/16 - 0s - loss: 3882.1638 - val_loss: 3966.3030 - 28ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 3827.4336 - val_loss: 3910.5894 - 31ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 3776.3860 - val_loss: 3862.7266 - 26ms/epoch - 2ms/step
Epoch 33/50
16/16 - 0s - loss: 3717.7732 - val_loss: 3805.0669 - 28ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 3663.1497 

16/16 - 0s - loss: 219.7842 - val_loss: 225.7230 - 29ms/epoch - 2ms/step
Epoch 18/50
16/16 - 0s - loss: 202.1947 - val_loss: 208.7763 - 29ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 186.5212 - val_loss: 195.7812 - 29ms/epoch - 2ms/step
Epoch 20/50
16/16 - 0s - loss: 172.2981 - val_loss: 181.3043 - 32ms/epoch - 2ms/step
Epoch 21/50
16/16 - 0s - loss: 162.3420 - val_loss: 169.1390 - 29ms/epoch - 2ms/step
Epoch 22/50
16/16 - 0s - loss: 152.3882 - val_loss: 163.4039 - 30ms/epoch - 2ms/step
Epoch 23/50
16/16 - 0s - loss: 145.7936 - val_loss: 152.5852 - 29ms/epoch - 2ms/step
Epoch 24/50
16/16 - 0s - loss: 137.0342 - val_loss: 147.2090 - 30ms/epoch - 2ms/step
Epoch 25/50
16/16 - 0s - loss: 132.0331 - val_loss: 141.4371 - 29ms/epoch - 2ms/step
Epoch 26/50
16/16 - 0s - loss: 126.6112 - val_loss: 135.8438 - 33ms/epoch - 2ms/step
Epoch 27/50
16/16 - 0s - loss: 122.9487 - val_loss: 131.4243 - 32ms/epoch - 2ms/step
Epoch 28/50
16/16 - 0s - loss: 119.5763 - val_loss: 133.4150 - 29ms/epoch - 2

Epoch 13/50
16/16 - 0s - loss: 4100.9668 - val_loss: 3749.3564 - 41ms/epoch - 3ms/step
Epoch 14/50
16/16 - 0s - loss: 3776.0505 - val_loss: 3433.8110 - 25ms/epoch - 2ms/step
Epoch 15/50
16/16 - 0s - loss: 3440.9810 - val_loss: 3134.8220 - 32ms/epoch - 2ms/step
Epoch 16/50
16/16 - 0s - loss: 3125.7292 - val_loss: 2828.6953 - 29ms/epoch - 2ms/step
Epoch 17/50
16/16 - 0s - loss: 2824.5530 - val_loss: 2561.7273 - 28ms/epoch - 2ms/step
Epoch 18/50
16/16 - 0s - loss: 2551.6895 - val_loss: 2288.3450 - 28ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 2284.9102 - val_loss: 2045.2170 - 33ms/epoch - 2ms/step
Epoch 20/50
16/16 - 0s - loss: 2036.0061 - val_loss: 1820.6978 - 23ms/epoch - 1ms/step
Epoch 21/50
16/16 - 0s - loss: 1812.4072 - val_loss: 1609.5859 - 29ms/epoch - 2ms/step
Epoch 22/50
16/16 - 0s - loss: 1597.4678 - val_loss: 1421.7446 - 28ms/epoch - 2ms/step
Epoch 23/50
16/16 - 0s - loss: 1403.8788 - val_loss: 1243.0461 - 35ms/epoch - 2ms/step
Epoch 24/50
16/16 - 0s - loss: 1208.1368 - 

Epoch 8/50
16/16 - 0s - loss: 719.6693 - val_loss: 796.4191 - 29ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 685.2028 - val_loss: 758.4371 - 34ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 656.3705 - val_loss: 726.2580 - 36ms/epoch - 2ms/step
Epoch 11/50
16/16 - 0s - loss: 630.1041 - val_loss: 695.3890 - 31ms/epoch - 2ms/step
Epoch 12/50
16/16 - 0s - loss: 604.8067 - val_loss: 664.6203 - 28ms/epoch - 2ms/step
Epoch 13/50
16/16 - 0s - loss: 580.8193 - val_loss: 636.7451 - 18ms/epoch - 1ms/step
Epoch 14/50
16/16 - 0s - loss: 558.3478 - val_loss: 611.8557 - 34ms/epoch - 2ms/step
Epoch 15/50
16/16 - 0s - loss: 537.7194 - val_loss: 587.3015 - 40ms/epoch - 2ms/step
Epoch 16/50
16/16 - 0s - loss: 519.6229 - val_loss: 562.0889 - 20ms/epoch - 1ms/step
Epoch 17/50
16/16 - 0s - loss: 498.2452 - val_loss: 542.7804 - 48ms/epoch - 3ms/step
Epoch 18/50
16/16 - 0s - loss: 481.9280 - val_loss: 525.0807 - 20ms/epoch - 1ms/step
Epoch 19/50
16/16 - 0s - loss: 463.8531 - val_loss: 503.2284 - 33ms

16/16 - 0s - loss: 133213.6719 - val_loss: 107410.2969 - 26ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 95352.5938 - val_loss: 75549.6875 - 31ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 67025.8672 - val_loss: 52189.7422 - 24ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 45965.7422 - val_loss: 34749.0352 - 24ms/epoch - 1ms/step
Epoch 7/50
16/16 - 0s - loss: 29449.5801 - val_loss: 20749.1875 - 27ms/epoch - 2ms/step
Epoch 8/50
16/16 - 0s - loss: 17016.3145 - val_loss: 12846.0312 - 28ms/epoch - 2ms/step
Epoch 9/50
16/16 - 0s - loss: 11458.9814 - val_loss: 10230.5000 - 34ms/epoch - 2ms/step
Epoch 10/50
16/16 - 0s - loss: 9665.6543 - val_loss: 9580.3770 - 25ms/epoch - 2ms/step
Epoch 11/50
16/16 - 0s - loss: 9136.0596 - val_loss: 9100.1973 - 29ms/epoch - 2ms/step
Epoch 12/50
16/16 - 0s - loss: 8639.4307 - val_loss: 8600.7900 - 28ms/epoch - 2ms/step
Epoch 13/50
16/16 - 0s - loss: 8176.5405 - val_loss: 8182.0381 - 30ms/epoch - 2ms/step
Epoch 14/50
16/16 - 0s - loss: 7805.7002 - va

Epoch 48/50
16/16 - 0s - loss: 526.0372 - val_loss: 474.2656 - 30ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 516.3776 - val_loss: 467.4972 - 16ms/epoch - 981us/step
Epoch 50/50
16/16 - 0s - loss: 507.1070 - val_loss: 460.3863 - 32ms/epoch - 2ms/step
10/10 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 9995.9043 - val_loss: 3781.2654 - 267ms/epoch - 17ms/step
Epoch 2/50
16/16 - 0s - loss: 2666.8484 - val_loss: 2780.5195 - 40ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 2168.8298 - val_loss: 2160.1089 - 28ms/epoch - 2ms/step
Epoch 4/50
16/16 - 0s - loss: 1662.4897 - val_loss: 1786.0107 - 31ms/epoch - 2ms/step
Epoch 5/50
16/16 - 0s - loss: 1416.6917 - val_loss: 1564.9781 - 28ms/epoch - 2ms/step
Epoch 6/50
16/16 - 0s - loss: 1220.5665 - val_loss: 1349.5496 - 33ms/epoch - 2ms/step
Epoch 7/50
16/16 - 0s - loss: 1049.3420 - val_loss: 1139.7667 - 31ms/epoch - 2ms/step
Epoch 8/50
16/16 - 0s - loss: 906.2698 - val_loss: 969.5132 - 31ms/epoch - 2ms/step
Ep

Epoch 43/50
16/16 - 0s - loss: 558.4084 - val_loss: 619.1453 - 31ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 550.5648 - val_loss: 610.5943 - 27ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 543.1832 - val_loss: 601.7810 - 29ms/epoch - 2ms/step
Epoch 46/50
16/16 - 0s - loss: 536.0353 - val_loss: 593.3849 - 28ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 529.2373 - val_loss: 585.3471 - 32ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 522.0543 - val_loss: 577.4260 - 31ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 515.6511 - val_loss: 569.7686 - 32ms/epoch - 2ms/step
Epoch 50/50
16/16 - 0s - loss: 509.5092 - val_loss: 561.8017 - 31ms/epoch - 2ms/step
10/10 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 18336.9883 - val_loss: 7391.9751 - 266ms/epoch - 17ms/step
Epoch 2/50
16/16 - 0s - loss: 3670.9133 - val_loss: 1357.4487 - 34ms/epoch - 2ms/step
Epoch 3/50
16/16 - 0s - loss: 1301.8905 - val_loss: 1409.4745 - 19ms/epoch - 1ms/step
Epoch 

Epoch 38/50
16/16 - 0s - loss: 339.7143 - val_loss: 388.9232 - 38ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 333.7736 - val_loss: 383.1469 - 22ms/epoch - 1ms/step
Epoch 40/50
16/16 - 0s - loss: 328.7160 - val_loss: 377.1609 - 29ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 323.0731 - val_loss: 372.4055 - 29ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 317.9797 - val_loss: 366.1463 - 33ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 313.1297 - val_loss: 361.1534 - 29ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 308.6733 - val_loss: 355.6028 - 27ms/epoch - 2ms/step
Epoch 45/50
16/16 - 0s - loss: 303.3179 - val_loss: 350.1273 - 30ms/epoch - 2ms/step
Epoch 46/50
16/16 - 0s - loss: 298.6321 - val_loss: 345.0697 - 36ms/epoch - 2ms/step
Epoch 47/50
16/16 - 0s - loss: 294.5644 - val_loss: 341.5366 - 25ms/epoch - 2ms/step
Epoch 48/50
16/16 - 0s - loss: 289.8836 - val_loss: 335.5575 - 34ms/epoch - 2ms/step
Epoch 49/50
16/16 - 0s - loss: 285.9314 - val_loss: 330.3574 - 25

Epoch 33/50
16/16 - 0s - loss: 534.7819 - val_loss: 718.5197 - 33ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 509.8805 - val_loss: 687.9046 - 30ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 485.7186 - val_loss: 657.4094 - 29ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 465.4688 - val_loss: 627.1675 - 29ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 444.3095 - val_loss: 602.2337 - 32ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 425.9303 - val_loss: 579.1748 - 32ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 408.6904 - val_loss: 553.6472 - 29ms/epoch - 2ms/step
Epoch 40/50
16/16 - 0s - loss: 392.0842 - val_loss: 532.3953 - 31ms/epoch - 2ms/step
Epoch 41/50
16/16 - 0s - loss: 377.1497 - val_loss: 512.3786 - 29ms/epoch - 2ms/step
Epoch 42/50
16/16 - 0s - loss: 362.7001 - val_loss: 491.0805 - 29ms/epoch - 2ms/step
Epoch 43/50
16/16 - 0s - loss: 348.3036 - val_loss: 474.4075 - 30ms/epoch - 2ms/step
Epoch 44/50
16/16 - 0s - loss: 335.7838 - val_loss: 452.5284 - 31

Epoch 28/50
16/16 - 0s - loss: 601.5952 - val_loss: 650.2379 - 31ms/epoch - 2ms/step
Epoch 29/50
16/16 - 0s - loss: 590.4077 - val_loss: 637.1619 - 31ms/epoch - 2ms/step
Epoch 30/50
16/16 - 0s - loss: 579.2130 - val_loss: 625.5493 - 32ms/epoch - 2ms/step
Epoch 31/50
16/16 - 0s - loss: 566.8392 - val_loss: 617.5998 - 31ms/epoch - 2ms/step
Epoch 32/50
16/16 - 0s - loss: 559.2255 - val_loss: 598.7637 - 30ms/epoch - 2ms/step
Epoch 33/50
16/16 - 0s - loss: 546.9405 - val_loss: 595.1703 - 31ms/epoch - 2ms/step
Epoch 34/50
16/16 - 0s - loss: 533.7672 - val_loss: 576.6267 - 29ms/epoch - 2ms/step
Epoch 35/50
16/16 - 0s - loss: 524.6815 - val_loss: 570.2245 - 30ms/epoch - 2ms/step
Epoch 36/50
16/16 - 0s - loss: 519.1809 - val_loss: 558.1859 - 32ms/epoch - 2ms/step
Epoch 37/50
16/16 - 0s - loss: 506.9787 - val_loss: 541.0843 - 28ms/epoch - 2ms/step
Epoch 38/50
16/16 - 0s - loss: 496.6577 - val_loss: 533.0673 - 29ms/epoch - 2ms/step
Epoch 39/50
16/16 - 0s - loss: 493.5835 - val_loss: 521.1349 - 29

Epoch 22/50
16/16 - 0s - loss: 419.3376 - val_loss: 410.6339 - 45ms/epoch - 3ms/step
Epoch 23/50
16/16 - 0s - loss: 350.9972 - val_loss: 342.5401 - 45ms/epoch - 3ms/step
Epoch 24/50
16/16 - 0s - loss: 296.7350 - val_loss: 285.0918 - 51ms/epoch - 3ms/step
Epoch 25/50
16/16 - 0s - loss: 254.8299 - val_loss: 240.1057 - 43ms/epoch - 3ms/step
Epoch 26/50
16/16 - 0s - loss: 225.5087 - val_loss: 209.9723 - 56ms/epoch - 4ms/step
Epoch 27/50
16/16 - 0s - loss: 202.1948 - val_loss: 187.7081 - 58ms/epoch - 4ms/step
Epoch 28/50
16/16 - 0s - loss: 184.3415 - val_loss: 169.3190 - 59ms/epoch - 4ms/step
Epoch 29/50
16/16 - 0s - loss: 172.4393 - val_loss: 156.8414 - 59ms/epoch - 4ms/step
Epoch 30/50
16/16 - 0s - loss: 162.6364 - val_loss: 145.4871 - 50ms/epoch - 3ms/step
Epoch 31/50
16/16 - 0s - loss: 156.7028 - val_loss: 142.9046 - 54ms/epoch - 3ms/step
Epoch 32/50
16/16 - 0s - loss: 148.9250 - val_loss: 130.6494 - 55ms/epoch - 3ms/step
Epoch 33/50
16/16 - 0s - loss: 145.3352 - val_loss: 128.8409 - 46

Epoch 16/50
16/16 - 0s - loss: 328.8680 - val_loss: 278.3730 - 42ms/epoch - 3ms/step
Epoch 17/50
16/16 - 0s - loss: 287.8977 - val_loss: 246.4832 - 28ms/epoch - 2ms/step
Epoch 18/50
16/16 - 0s - loss: 252.6650 - val_loss: 213.8636 - 31ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 222.2687 - val_loss: 187.6342 - 24ms/epoch - 2ms/step
Epoch 20/50
16/16 - 0s - loss: 196.6196 - val_loss: 174.0537 - 30ms/epoch - 2ms/step
Epoch 21/50
16/16 - 0s - loss: 176.0712 - val_loss: 148.3936 - 31ms/epoch - 2ms/step
Epoch 22/50
16/16 - 0s - loss: 157.3916 - val_loss: 137.1350 - 32ms/epoch - 2ms/step
Epoch 23/50
16/16 - 0s - loss: 145.0163 - val_loss: 123.9476 - 44ms/epoch - 3ms/step
Epoch 24/50
16/16 - 0s - loss: 139.2392 - val_loss: 134.9471 - 23ms/epoch - 1ms/step
Epoch 25/50
16/16 - 0s - loss: 131.3638 - val_loss: 109.2775 - 48ms/epoch - 3ms/step
Epoch 26/50
16/16 - 0s - loss: 121.2233 - val_loss: 104.3287 - 39ms/epoch - 2ms/step
Epoch 27/50
16/16 - 0s - loss: 114.3278 - val_loss: 101.3351 - 39

Epoch 11/50
16/16 - 0s - loss: 503.2734 - val_loss: 461.3611 - 39ms/epoch - 2ms/step
Epoch 12/50
16/16 - 0s - loss: 443.4716 - val_loss: 416.2594 - 29ms/epoch - 2ms/step
Epoch 13/50
16/16 - 0s - loss: 400.7083 - val_loss: 383.9135 - 27ms/epoch - 2ms/step
Epoch 14/50
16/16 - 0s - loss: 358.7148 - val_loss: 344.5515 - 27ms/epoch - 2ms/step
Epoch 15/50
16/16 - 0s - loss: 325.7497 - val_loss: 316.9745 - 31ms/epoch - 2ms/step
Epoch 16/50
16/16 - 0s - loss: 296.4010 - val_loss: 292.6446 - 20ms/epoch - 1ms/step
Epoch 17/50
16/16 - 0s - loss: 273.9265 - val_loss: 269.6182 - 31ms/epoch - 2ms/step
Epoch 18/50
16/16 - 0s - loss: 253.3138 - val_loss: 251.4854 - 32ms/epoch - 2ms/step
Epoch 19/50
16/16 - 0s - loss: 236.6231 - val_loss: 231.4029 - 31ms/epoch - 2ms/step
Epoch 20/50
16/16 - 0s - loss: 221.1570 - val_loss: 216.5264 - 30ms/epoch - 2ms/step
Epoch 21/50
16/16 - 0s - loss: 205.8788 - val_loss: 204.2527 - 30ms/epoch - 2ms/step
Epoch 22/50
16/16 - 0s - loss: 193.9928 - val_loss: 188.6345 - 37

In [56]:
# Report the mean and the standard deviation of the mean squared errors.
print("Mean of MSE : ",np.mean(MS_Error))
print("SD of MSE :", np.std(MS_Error))

Mean of MSE :  487.09447994389933
SD of MSE : 517.3872045307525


<strong>You can refer to this [link](https://keras.io/models/sequential/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDL0101ENSkillsNetwork20718188-2022-01-01) to learn about other functions that you can use for prediction or evaluation.</strong>


Feel free to vary the following and note what impact each change has on the model's performance:

1.  Increase or decreate number of neurons in hidden layers
2.  Add more hidden layers
3.  Increase number of epochs


### Thank you for completing this lab!

This notebook was created by [Alex Aklson](https://www.linkedin.com/in/aklson/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDL0101ENSkillsNetwork20718188-2022-01-01). I hope you found this lab interesting and educational. Feel free to contact me if you have any questions!


## Change Log

| Date (YYYY-MM-DD) | Version | Changed By | Change Description                                          |
| ----------------- | ------- | ---------- | ----------------------------------------------------------- |
| 2020-09-21        | 2.0     | Srishti    | Migrated Lab to Markdown and added to course repo in GitLab |

<hr>

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>


This notebook is part of a course on **Coursera** called *Introduction to Deep Learning & Neural Networks with Keras*. If you accessed this notebook outside the course, you can take this course online by clicking [here](https://cocl.us/DL0101EN_Coursera_Week3\_LAB1).


<hr>

Copyright © 2019 [IBM Developer Skills Network](https://cognitiveclass.ai/?utm_medium=dswb&utm_source=bducopyrightlink&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDL0101ENSkillsNetwork20718188-2022-01-01&utm_campaign=bdu). This notebook and its source code are released under the terms of the [MIT License](https://bigdatauniversity.com/mit-license/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDL0101ENSkillsNetwork20718188-2022-01-01).
